src/wms/src/main/java/org/geoserver/wms/legendgraphic/LegendMerger.java

```diff
@@ -0,0 +1,662 @@
+/* (c) 2016 Open Source Geospatial Foundation - all rights reserved
+ * This code is licensed under the GPL 2.0 license, available at the root
+ * application directory.
+ */
+
+package org.geoserver.wms.legendgraphic;
+
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.Graphics2D;
+import java.awt.RenderingHints;
+import java.awt.image.BufferedImage;
+import java.awt.image.IndexColorModel;
+import java.awt.image.RenderedImage;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import org.geoserver.wms.GetLegendGraphicRequest;
+import org.geoserver.wms.legendgraphic.LegendUtils.LegendLayout;
+import org.geoserver.wms.map.ImageUtils;
+import org.geotools.styling.Description;
+import org.geotools.styling.Rule;
+import org.opengis.util.InternationalString;
+
+public class LegendMerger {

```

Comment:
Class javadoc?

Was going to suggest this class be used as param object be passed into mergeRasterLegends, but then noticed that this class is acting as a facade for the Column, Row, and build methods. A quick javadoc can save developers time.